src/bootman/kernel.c

```diff
@@ -213,7 +213,7 @@ Kernel *boot_manager_inspect_kernel(BootManager *self, char *path)
                     string_printf("initrd-%s.%s.%s-%d", KERNEL_NAMESPACE, type, version, release);
         }
 
-        kern->meta.release = (int16_t)release;
+        kern->meta.release = (int32_t)release;
```

Comment:
This fixes the issue. What I do not understand however is why you cast it to `int32_t` instead of using `int` directy. It is defined [here](https://github.com/clearlinux/clr-boot-manager/blob/master/src/bootman/bootman.h#L54) as an `int` after all?

*note:* I did try to change your patch to cast to an `int` and it worked equally well.
```suggestion
        kern->meta.release = (int)release;
```